### ISSUE TITLE
False positive for `arguments-differ` when collecting all arguments

### ISSUE THREAD

**Issue by benf-wspdigital** [PyCQA/pylint#1553]
The `arguments-differ` check will complain about this common pattern:

```
class Ipsum:
    def dolor(self, elit):
        pass


class LoremIpsum(Ipsum):
    def dolor(self, *args, **kwargs):
        super().dolor(*args, **kwargs)
```

PyLint reports:

```
$ python3 -m pylint --reports=n --enable=arguments-differ lorem.py 
************* Module lorem
lorem.py:7: [W0235(useless-super-delegation), LoremIpsum.dolor] Useless super delegation in method 'dolor'
lorem.py:7: [W0221(arguments-differ), LoremIpsum.dolor] Parameters differ from overridden 'dolor' method
```

That code should not trigger the `arguments-differ` check, because that pattern is explicitly just passing through the arguments to the superclass's method.

**Comment by AWhetter** [comment]
To any beginner contributors looking to fix this: `arguments-differ` is raised around https://github.com/PyCQA/pylint/blob/fbdf5ba095ebbbf04db12f6c3e88e21152a30507/pylint/checkers/classes.py#L1233
This will likely need a fix in `_different_parameters()` or `_has_different_parameters()` where we should add a check for *args, and **kwargs in `overridden`.

**Comment by kczapla** [comment]
Hi,
I made simple fix here #1674.

**Comment by GergelyKalmar** [comment]
This fix doesn't seem to work when the overridden methods have default values:

```
class Ipsum:
    def dolor(self, elit=None):
        pass
```

```
pylint lorem.py
...
lorem.py:7:4: W0222: Signature differs from overridden 'dolor' method (signature-differs)
```

Is this to be expected?

**Comment by Pierre-Sassoulas** [comment]
@GergelyKalmar I don't think the example you gave has all the code it should, we can't know what is inherited.

**Comment by GergelyKalmar** [comment]
@Pierre-Sassoulas Sorry, I meant it in the same context as the original example in the issue report, just with a default value added:

```
class Ipsum:
    def dolor(self, elit=None):
        pass


class LoremIpsum(Ipsum):
    def dolor(self, *args, **kwargs):
        super().dolor(*args, **kwargs)
```

This should fail with a ``signature-differs`` error.

**Comment by Pierre-Sassoulas** [comment]
Thank you. It seems like a bug, I think you can open a new issue with this example :)
